pkg/apis/dweller/controller.go

```diff
@@ -0,0 +1,197 @@
+package dweller
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	dweller_v1a1 "github.com/fukt/dweller/pkg/apis/dweller/v1alpha1"
+)
+
+const maxRetries = 5
+
+type Controller struct {
+	logger    *logrus.Entry
+	clientset kubernetes.Interface
+	queue     workqueue.RateLimitingInterface
+	informer  cache.SharedIndexInformer
+}
+
+func Start() {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		fmt.Printf("error creating k8s client: %s", err.Error())
+		return
+	}
+
+	kubeClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		fmt.Printf("error creating k8s client: %s", err.Error())
+		return
+	}
+
+	c := newController(config, kubeClient)
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+
+	go c.Run(stopCh)
+
+	sigterm := make(chan os.Signal, 1)
+	signal.Notify(sigterm, syscall.SIGTERM)
+	signal.Notify(sigterm, syscall.SIGINT)
+	<-sigterm
+}
+
+func newController(k8sConfig *rest.Config, client kubernetes.Interface) *Controller {
+	queue := workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
+
+	config := *k8sConfig
+	gv := dweller_v1a1.SchemeGroupVersion
+	config.APIPath = "/apis"
+	config.GroupVersion = &gv
+	config.NegotiatedSerializer = serializer.DirectCodecFactory{CodecFactory: dweller_v1a1.CodecFactory}
+	c, err := rest.RESTClientFor(&config)
+	if nil != err {
+		fmt.Printf("error creating client: %v", err)
+		return nil
+	}
+
+	informer := cache.NewSharedIndexInformer(
```

Comment:
orly? did ingress authors suppose that it would be used only by them? too strict assumption, imo

Reply:
> did ingress authors suppose that it would be used only by them?

I think it is not a correct example. *ingress* is a resource that is made to be used by different controllers. But our vault secret claim resource is made to be used only by our controller. If it is not, there is some flaw in design or something.